artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/VersionedStompFrameHandler.java

```diff
@@ -344,7 +348,8 @@ public StompFrame createMessageFrame(ICoreMessage serverMessage, StompSubscripti
       }
       frame.setByteBody(data);
 
-      StompUtils.copyStandardHeadersFromMessageToFrame((serverMessage), frame, deliveryCount);
+      frame.addHeader(Stomp.Headers.Message.MESSAGE_ID, String.valueOf(consumer.getID()) + StompSession.MESSAGE_ID_SEPARATOR + String.valueOf(serverMessage.getMessageID()));
```

Comment:
This is going to be done for every message dispatched from the broker so it would be worth making this is as fast as possible. Perhaps something like:
```java
frame.addHeader(Stomp.Headers.Message.MESSAGE_ID, new StringBuilder(32).append(consumer.getID()).append(StompSession.MESSAGE_ID_SEPARATOR).append(serverMessage.getMessageID()).toString());
```

Reply:
Done, but I set initial StringBuilder capacity to 41 to fit two long values and separator.

Reply:
How did you arrive at the `41` value?

Reply:
Minimum long value is 20 characters (-9223372036854775808), so two long values and separator could require 41 characters total.

Reply:
Message and consumer IDs should never be < 0 so the potential values are 0 (1 character) through 9223372036854775807 (19 characters). Since these characters are just digits (i.e. part of ISO-8859-1) then they will be represented by a single byte in the `byte[]` that backs the `String`. We want to avoid expanding the backing array but we also want to avoid over-allocating (and therefore wasting) space. Ten characters for each long seems like a fair middle ground.